starknet-replay/src/histogram/mod.rs

```diff
@@ -0,0 +1,281 @@
+        Ok(u32::try_from(x_label_area_size)?)
+    }
+
+    /// Calculate the maximum number shown on the y-axis of the histogram.
+    ///
+    /// The principle is to automatically resize the axis depending on the
+    /// number of times the most frequently called `libfunc` is called. This
+    /// number is rounded to the next hundreds.
+    ///
+    /// # Arguments
+    ///
+    /// - `max_frequency`: The highest frequency to be represented in the
+    ///   histogram.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if there is an overflow in the calculation of the y-axis
+    /// extension.
+    fn calc_max_y_axis(max_frequency: usize) -> Result<usize, HistogramError> {
+        // `div` and `add` don't need to be checked because they will always return a
+        // number less than `max_frequency`, therefore fitting in the size of `usize`.
+        let max_y_axis = max_frequency.div(100).add(1).checked_mul(100);
+        max_y_axis.ok_or(HistogramError::MathOverflow("calc_max_y_axis".to_string()))
+    }
+
+    /// Calculate the width in pixels of the SVG image containing the histogram.
+    ///
+    /// The idea is to resize the width depending on the number of buckets to be
+    /// plotted plus some margin.
+    ///
+    /// # Arguments
+    ///
+    /// - `number_of_buckets`: The number of buckets in the histogram.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if there is an overflow in the calculation of the width.
+    fn calc_width(number_of_buckets: usize) -> Result<Pixel, HistogramError> {
+        let number_of_buckets: Pixel = u32::try_from(number_of_buckets)?;
+        let pixels_per_bucket: Pixel = 40;
+        let extra_margins: Pixel = 250;
+        number_of_buckets
+            .checked_mul(pixels_per_bucket)
+            .and_then(|n| n.checked_add(extra_margins))
+            .ok_or(HistogramError::MathOverflow("calc_width".to_string()))
+    }
+
+    /// Calculate the height in pixels of the SVG image containing the
+    /// histogram.
+    ///
+    /// The idea is to provide 2 pixels of height for buckets with frequency of
+    /// 1 plus some margin at the bottom for the labels of the x-axis.
+    ///
+    /// # Arguments
+    ///
+    /// - `max_y_axis`: The max extension of the y-axis.
+    /// - `x_axis_label_space`: The amount of margin for x-axis labels.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if there is an overflow in the calculation of the
+    /// height.
+    fn calc_height(max_y_axis: usize, x_axis_label_space: Pixel) -> Result<Pixel, HistogramError> {
+        let max_y_axis: Pixel = u32::try_from(max_y_axis)?;
+        let pixels_for_each_call: Pixel = 2;
+        max_y_axis
+            .checked_mul(pixels_for_each_call)
+            .and_then(|n| n.checked_add(x_axis_label_space))
+            .ok_or(HistogramError::MathOverflow("calc_height".to_string()))
+    }
+}
+
+/// This function generates and saves the libfunc frequency histogram.
```

Comment:
I think it makes sense to separate rendering and saving. Working with the histogram computationally may be desirable!